RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -74,53 +80,84 @@ public bool Generate(NuGetReference analyzeRef, string language, string sqaleFil
                 throw new ArgumentNullException("language");
             }
             SupportedLanguages.ThrowIfNotSupported(language);
+            if (string.IsNullOrWhiteSpace(outputDirectory))
+            {
+                throw new ArgumentNullException("outputDirectory");
+            }
+
+            IPackage package = TryGetPackage(analyzerRef);
```

Comment:
The changes from here to line 121 are not SQALE-related. I've cleaned up the code slightly by moving a couple of bits into helper methods, and changing the order of operations to reduce unnecessary work e.g. there's no point in populating the plugin definition if we can't find any analyzers, so we should look for the analyzers first.